node/libs/roles/src/attester/messages/msg.rs

```diff
@@ -0,0 +1,279 @@
+        self.0.none()
+    }
+}
+
+/// A struct that represents a set of attesters. It is used to store the current attester set.
+/// We represent each attester by its attester public key.
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+pub struct Committee {
+    vec: Vec<WeightedAttester>,
+    indexes: BTreeMap<PublicKey, usize>,
+    total_weight: u64,
+}
+
+impl Committee {
+    /// Creates a new Committee from a list of attester public keys.
+    pub fn new(attesters: impl IntoIterator<Item = WeightedAttester>) -> anyhow::Result<Self> {
+        let mut weighted_attester = BTreeMap::new();
+        let mut total_weight: u64 = 0;
+        for attester in attesters {
+            anyhow::ensure!(
+                !weighted_attester.contains_key(&attester.key),
+                "Duplicated attester in attester Committee"
+            );
+            anyhow::ensure!(
+                attester.weight > 0,
+                "Attester weight has to be a positive value"
+            );
+            total_weight = total_weight
+                .checked_add(attester.weight)
+                .context("Sum of weights overflows in attester Committee")?;
+            weighted_attester.insert(attester.key.clone(), attester);
+        }
+        anyhow::ensure!(
+            !weighted_attester.is_empty(),
+            "Attester Committee must contain at least one attester"
+        );
+        Ok(Self {
+            vec: weighted_attester.values().cloned().collect(),
+            indexes: weighted_attester
+                .values()
+                .enumerate()
+                .map(|(i, v)| (v.key.clone(), i))
+                .collect(),
+            total_weight,
+        })
+    }
+
+    /// Iterates over weighted attesters.
+    pub fn iter(&self) -> impl Iterator<Item = &WeightedAttester> {
+        self.vec.iter()
+    }
+
+    /// Iterates over attester keys.
+    pub fn iter_keys(&self) -> impl Iterator<Item = &PublicKey> {
+        self.vec.iter().map(|v| &v.key)
+    }
+
+    /// Returns the number of attesters.
+    #[allow(clippy::len_without_is_empty)] // a valid `Committee` is always non-empty by construction
+    pub fn len(&self) -> usize {
+        self.vec.len()
+    }
+
+    /// Returns true if the given attester is in the attester committee.
+    pub fn contains(&self, attester: &PublicKey) -> bool {
+        self.indexes.contains_key(attester)
+    }
+
+    /// Get attester by its index in the committee.
+    pub fn get(&self, index: usize) -> Option<&WeightedAttester> {
+        self.vec.get(index)
+    }
+
+    /// Get the index of a attester in the committee.
+    pub fn index(&self, attester: &PublicKey) -> Option<usize> {
+        self.indexes.get(attester).copied()
+    }
+
+    /// Computes the leader for the given view.
+    pub fn view_leader(&self, view_number: ViewNumber) -> PublicKey {
+        let index = view_number.0 as usize % self.len();
+        self.get(index).unwrap().key.clone()
+    }
```

Comment:
there is no such a thing as leader in an attester committee

Reply:
+1